### ISSUE TITLE
4.1.0 indent regression on jsx binary expressions when using parenthesis

### ISSUE THREAD

**Issue by lukeapage** [eslint/eslint#8832]
**Tell us about your environment**

* **ESLint Version:** 4.1.1
* **Node Version:** 6.9.1
* **npm Version:** 4.5.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
        "indent": [
            "error",
            4,
            {
                "SwitchCase": 1
            }
        ],
```

**What did you do? Please include the actual source code causing the issue.**

```js
function A() {
    return (
        <div>
            {
                b && (
                    <div>
                    </div>
                )
            }
        </div>
    );
}
```

**What did you expect to happen?**

No errors - this code is fine in 4.0

**What actually happened? Please include the actual, raw output from ESLint.**

indentation errors:

![image](https://user-images.githubusercontent.com/309321/27653183-8d9abecc-5c3e-11e7-88d1-8ff00551b0cd.png)

passing code:

```js
function A() {
    return (
        <div>
            {
                b && (
                <div>
                </div>
            )
            }
        </div>
    );
}
```

Note it seems that without parenthesis it works fine.. this code passes..

```js
function A() {
    return (
        <div>
            {
                b &&
                    <div>
                    </div>
            }
        </div>
    );
}
```

**Comment by not-an-aardvark** [comment]
Thanks for reporting, I can reproduce this.

**Comment by lukeapage** [comment]
Thanks @not-an-aardvark and thanks in general for the great work on indent, it's very much appreciated
